lib/mix_unused/filter.ex

```diff
@@ -24,23 +24,41 @@ defmodule MixUnused.Filter do
   @spec reject_matching(exports :: Exports.t(), patterns :: [pattern()]) ::
           Exports.t()
   def reject_matching(exports, patterns) do
-    filters =
-      Enum.map(patterns, fn
-        {_m, _f, _a} = entry -> entry
-        {m, f} -> {m, f, :_}
-        {m} -> {m, :_, :_}
-        m when is_atom(m) -> {m, :_, :_}
-        %Regex{} = m -> {m, :_, :_}
-        cb when is_function(cb) -> cb
-      end)
+    filters = normalize_filter_patterns(patterns)
 
     Enum.reject(exports, fn {func, meta} ->
       Enum.any?(filters, &mfa_match?(&1, func, meta))
     end)
     |> Map.new()
   end
 
-  @spec mfa_match?(mfa(), pattern(), Meta.t()) :: boolean()
+  @doc """
+  Get values in `exports` that match any pattern in `patterns`
+  """
```

Comment:
I would not write this doc: the @moduledoc is set to false (so it is not show with `mix docs`) and it is an internal module.